src/adapters/fauna/index.js

```diff
@@ -0,0 +1,420 @@
+// create faunadb server key
+// create collections: users, accounts, sessions, verificationRequests
+// create required indexes
+
+import faunadb, { query as q } from 'faunadb'
+import { v4 as uuidv4 } from 'uuid'
+import { createHash, randomBytes } from 'crypto'
+
+const INDEX_USERS_ID = 'index_users_id'
+const INDEX_USERS_EMAIL = 'index_users_email'
+const INDEX_USERS_ACCOUNT_ID_PROVIDER_ID = 'index_accounts_providerId_providerAccountId'
+const INDEX_VERIFICATION_REQUESTS_TOKEN = 'index_verificationRequests_token'
+const INDEX_SESSIONS_ID = 'index_sessions_id'
+const INDEX_SESSIONS_SESSION_TOKEN = 'index_sessions_sessionToken'
+
+const serverClient = new faunadb.Client({ secret: process.env.FAUNADB_SERVER_KEY })
+
+function faunaWrapper (faunaQ, errorTag) {
+  try {
+    return serverClient.query(faunaQ)
+  } catch (error) {
+    console.error(errorTag, error)
+    return Promise.reject(new Error(errorTag, error))
+  }
+}
+
+const Adapter = (config, options = {}) => {
+  async function getAdapter (appOptions) {
+    if (appOptions && (!appOptions.session || !appOptions.session.maxAge)) {
+      console.log('no default options for session')
+    }
+
+    const defaultSessionMaxAge = 30 * 24 * 60 * 60 * 1000
+    const sessionMaxAge =
+            appOptions && appOptions.session && appOptions.session.maxAge
+              ? appOptions.session.maxAge * 1000
+              : defaultSessionMaxAge
+    const sessionUpdateAge =
+            appOptions && appOptions.session && appOptions.session.updateAge
```

Comment:
Noticed this while testing with 0, was using the default.

``appOptions && appOptions.session && (appOptions.session.updateAge || appOptions.session.updateAge === 0)``

Reply:
The logic is kept in sync with other adapters: https://github.com/nextauthjs/next-auth/tree/main/src/adapters
Can you raise an issue for this?

Reply:
checked master, and both prisma and typeorm adapters have the right logic (fallback to 0). 

Am I missing something here?
```    
const sessionUpdateAge = (appOptions && appOptions.session && appOptions.session.updateAge)
      ? appOptions.session.updateAge * 1000
      : 0
```